spec/lib/zadarest/client_spec.rb

```diff
@@ -0,0 +1,113 @@
+describe Zadarest::Client do
+  it 'gets the api_endpoint from the config object' do
+    api_endpoint = 'http://example.com'
+    config = Zadarest::Config.new(api_endpoint: api_endpoint)
+    allow(Faraday).to receive(:new)
+
+    described_class.new(config: config)
+
+    expect(Faraday).to have_received(:new).with(url: api_endpoint)
+  end
+
+  context 'get' do
+    it 'uses the faraday client correctly' do
+      faraday_double = get_faraday_double
+      url = '/foobar'
+      params = { param1: 1 }
+
+      config = Zadarest::Config.new(access_key: access_key)
+      client = described_class.new(config: config)
+      client.get(url: url, params: params)
+
+      expect(faraday_double).to have_received(:get).with(
+        url, params, default_faraday_headers
+      )
+    end
+  end
+
+  context 'post' do
+    it 'uses the faraday client correctly' do
+      faraday_double = get_faraday_double
+      url = '/foobar'
+      params = { param1: 1 }
+      json_encoded_params = params.to_json
+
+      config = Zadarest::Config.new(access_key: access_key)
+      client = described_class.new(config: config)
+      client.post(url: url, params: params)
+
+      expect(faraday_double).to have_received(:post).with(
+        url, json_encoded_params, default_faraday_headers
+      )
+    end
+
+  end
+
+  context 'delete' do
+    it 'uses the faraday client correctly' do
+      faraday_double = get_faraday_double
+      url = '/foobar'
+
+      config = Zadarest::Config.new(access_key: access_key)
+      client = described_class.new(config: config)
+      client.delete(url: url)
+
+      expect(faraday_double).to have_received(:delete).with(
+        url, {}, default_faraday_headers
+      )
+    end
+  end
+
+  it 'parses the returned json packet' do
```

Comment:
These could probably be extracted into shared examples and tested against all the supported methods.